src/components/articles/createArticle.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import NewArticle from "../../views/articles/createArticle";
+import { connect } from "react-redux";
+import {
+  createNewArticleAction,
+  publishNewArticleAction
+} from "../../actions/articleActions/articleAction";
+import PropTypes from "prop-types";
+import { myHeaders } from "../../utils/myHeaders";
+import { toast } from "react-toastify";
+
+class CreateArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      allcategory: [],
+      title: "",
+      description: "",
+      body: "",
+      category: "",
+      tags: []
+    };
+    this.handleEditorChange = this.handleEditorChange.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.handleSelectChange = this.handleSelectChange.bind(this);
+    this.handlePublish = this.handlePublish.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.categories();
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  };
+
+  handleSelectChange = e => {
+    this.setState({ category: e.target.value });
+  };
+
+  handlePublish = e => {
+    e.preventDefault();
+    this.props.dispatch(
+      publishNewArticleAction(this.props.articles.article.slug)
+    );
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const data = {
+      article: {
+        title: this.state.title,
+        description: this.state.description,
+        body: this.state.body,
+        tags: this.state.tags,
+        category: this.state.category
+      }
+    };
+    this.props.dispatch(createNewArticleAction(data));
+  };
+
+  handleEditorChange = value => {
+    this.setState({ body: value });
+  };
+
+  categories = () => {
+    let initial = [];
+    fetch(
+      "https://ah-backend-lannister-staging.herokuapp.com/api/categories/",
+      {
+        method: "GET",
+        headers: myHeaders
+      }
+    )
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        initial = data.categorys.results.map(categ => {
```

Comment:
Need for better naming i.e instead use categories and category instead of categ or something along those lines

Reply:
Sure, i have used category instead of categ. but for the *categorys*, its the object return from the backend and renaming would mean modifying the backend first.
![image](https://user-images.githubusercontent.com/37612799/48248547-d63d1a00-e408-11e8-8d88-5554677f3132.png)